### PR TITLE
Update indexing to include scsb in browse lists

### LINKED ISSUE #1790: In traject include scsb records in call_number_browse_s and call_number_display
We reviewed with @mzelesky traject and we need to consider scsb records for call_number_browse_s and call_number_display. This will allow for keyword search of the call _numbers and will allow for call_number browse.

### PR DESCRIPTION
**Pull request by carolyncole**
fixes #1790

### CODE REVIEW

File: spec/marc_to_solr/lib/config_spec.rb [hunk 1]
```diff
@@ -251,7 +251,10 @@ def fixture_record(fixture_name)
       expect(@record_call_number2['call_number_browse_s']).to eq(["CD- 50000"])
     end
     it 'returns the enrichment call_number_browse with k at the end' do
-      expect(@sample41['call_number_browse_s']).to eq(["GA 2015.00160 A Middle 30/Drawer 11/GC024/Full Folio/20th-21st c./Artists A"])
+      expect(@sample41['call_number_browse_s']).to contain_exactly("GA 2015.00160", "GA 2015.00160 A Middle 30/Drawer 11/GC024/Full Folio/20th-21st c./Artists A")
```
Comment (carolyncole) [inline spec/marc_to_solr/lib/config_spec.rb:254]: @christinach I believe the `GA 2015.00160` being repeated here is unexpected: https://catalog.princeton.edu/catalog/99106471643506421 There is only one item.  It does not have an 8 field, but neither does the SCSB record.

File: marc_to_solr/lib/princeton_marc.rb [hunk 2]
```diff
@@ -641,10 +641,18 @@ def alma_code_start_53?(code)
   code.to_s.start_with?("53") && code.to_s.end_with?("06421")
 end
 
+def scsb_code_start?(code)
```
Comment (carolyncole) [inline marc_to_solr/lib/princeton_marc.rb:644]: I had to add this filter of we ended up getting another call number for https://catalog.princeton.edu/catalog/99106471643506421, which is @sample41 in the tests.

File: marc_to_solr/lib/princeton_marc.rb [hunk 3]
```diff
@@ -641,10 +641,18 @@ def alma_code_start_53?(code)
...
+  code.to_s.start_with?("scsb")
+end
+
 def alma_852(record)
   record.fields('852').select { |f| alma_code_start_22?(f['8']) }
 end
 
+def alma_and_scsb_852(record)
+  record.fields('852').select { |f| alma_code_start_22?(f['8']) || scsb_code_start?(f['b']) }
+end
```
Comment (christinach) [inline marc_to_solr/lib/princeton_marc.rb:654]: We have a check for scsb in princeton_marc, which is to check the value of the document id. SCSB starts with `SCSB-`, `scsb_doc?(record['001'].value)`. 
@mzelesky is it a robust check using subfield $b to check for `scsb` text? If yes then I think it's ok to have one more check besides the id.

Comment (mzelesky) [reply marc_to_solr/lib/princeton_marc.rb:654]: I think it's good to check the subfield b of an 852 for SCSB records. However, I think that the 'alma_and_scsb_852' needs to be separate checks. So, like we have 'alma_852', we should also have 'scab_852'. Then, if the record is identified as a SCSB record, go the scsb_852 route. If the record is identified as an All record, go the alma_852 route.

Comment (carolyncole) [reply marc_to_solr/lib/princeton_marc.rb:654]: @mzelesky is it possible to have a record that has both recap and non recap 852 fields?  I was unsure, which is why I put them together.

**Review by christinach — APPROVED** [review]

**Review by christinach — APPROVED** [review]
@carolyncole thank you

**Pull request merged**